test/OrchardCore.Tests/Recipes/RecipeExecutorTests.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using Moq;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Builders;
+using OrchardCore.Environment.Shell.Scope;
+using OrchardCore.Recipes.Events;
+using OrchardCore.Recipes.Models;
+using OrchardCore.Recipes.Services;
+using OrchardCore.Scripting;
+using Xunit;
+
+namespace OrchardCore.Recipes
+{
+    public class RecipeExecutorTests
+    {
+        [Theory]
+        [InlineData("recipe1", "[locale en]You have successfully registered![/locale][locale fr]Vous vous êtes inscrit avec succès![/locale]")]
+        [InlineData("recipe2", "[1js: valiables('now')]")]
+        [InlineData("recipe3", "js: valiables('now')")]
+        [InlineData("recipe4", "[locale en]This text contain a colon ':' symbol[/locale][locale fr]Ce texte contient un symbole du côlon ':'[/locale]")]
+        [InlineData("recipe5", "[sc text='some : text'/]")]
+        [InlineData("recipe6", @"<code>
+[
+	{
+		color: ""red"",
+		value: ""#f00""
+	}
+]
+</code>")]
+        public async Task RemoveBracketsIfTheScriptExpressionProcessed(string recipeName, string expected)
+        {
+            // Arrange
+            var shellHostMock = new Mock<IShellHost>();
+            var shellSettings = new ShellSettings { Name = "Test" };
+            var services = new ServiceCollection()
+                .AddScripting();
+            shellHostMock.Setup(h => h.GetScopeAsync(It.IsAny<ShellSettings>()))
+                .Returns(Task.FromResult(new ShellScope(new ShellContext {
+                    ServiceProvider = services.BuildServiceProvider(),
+                    Settings = shellSettings
+                })));
+            var recipeEventHandlers = new List<IRecipeEventHandler> { new RecipeEventHandler() };
+            var loggerMock = new Mock<ILogger<RecipeExecutor>>();
+            var recipeExecutor = new RecipeExecutor(shellHostMock.Object, shellSettings, recipeEventHandlers, loggerMock.Object);
+            var executionId = Guid.NewGuid().ToString("n");
+            var recipeDescriptor = new RecipeDescriptor { RecipeFileInfo = GetRecipeFileInfo(recipeName) };
+
+            // Act
+            try
```

Comment:
why would you try catch this?

Reply:
Regarding this I forgot to tell you guys that execute throw an exception for accessing a disposable object `IServiceProvider` but it works, if someone can fix this, will be better